rewrite-gradle/src/test/java/org/openrewrite/gradle/ChangeDependencyClassifierTest.java

```diff
@@ -411,4 +412,84 @@ void noNewClassifier_2(String group, String artifact) {
           )
         );
     }
+
+    @Test
+    void worksWithDependencyDefinedInJvmTestSuite() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeDependencyClassifier("org.openrewrite", "*", "classified", ""))
+            .beforeRecipe(withToolingApi()),
```

Comment:
The `withToolingApi()` will need to be put in the defaults `beforeRecipe` and will no longer be needed in each method case across all of the unit tests.

Presently, the other test cases are all passing because the code is falling back to the `MethodMatcher`. With my other comments, this won't be the case anymore and we will need the `GradleProject` marker for the all of these other tests for the recipes being updated in this PR.

Reply:
Do we expect any issues with this for tests downstream looking at rewrite-java-dependencies and beyond? Just wondering what work might lie ahead when we merge this; it's no problem, but good to be aware.

Reply:
So that's one benefit of retaining the `MethodMatcher` as a fallback throughout these recipes (partly what I intend to polish tomorrow for the others). In this way, we will use the `GradleDependency` trait when we have the `GradleProject` marker -- which will guarantee the configuration is registered with Gradle, including previously unknown ones -- and will fallback to our internalized list of configurations found in `DependencyHandlerSpec` when such a marker cannot be found.